### PR TITLE
Raise a more meaningful error message when React server is booting up or down

### PR DESCRIPTION
**Pull request by ismail-syed**
## Description

Fixes https://github.com/Shopify/quilt/issues/837

Screen Shot 2019-08-20 at 4 59 35 PM

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above

### CODE REVIEW

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb [hunk 1]
```diff
@@ -8,11 +8,16 @@ def render_react
...
-        callbacks.on_response do |status_code, _response|
-          ReactRenderable.log("[ReactRenderable] #{url} returned #{status_code}")
+      begin
+        reverse_proxy(url) do |callbacks|
+          callbacks.on_response do |status_code, _response|
+            ReactRenderable.log("[ReactRenderable] #{url} returned #{status_code}")
+          end
         end
+      rescue Errno::ECONNREFUSED => e
+        raise "[ReactRenderable] Errno::ECONNREFUSED: Waiting for React server to boot up. If this error presists verify that @shopify/react-server is configured on #{url}"
```
Comment (ismail-syed) [inline gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: Should I drop the `[ReactRenderable]` prefixing

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb [hunk 2]
```diff
@@ -8,11 +8,16 @@ def render_react
...
-      reverse_proxy(url) do |callbacks|
-        callbacks.on_response do |status_code, _response|
-          ReactRenderable.log("[ReactRenderable] #{url} returned #{status_code}")
+      begin
+        reverse_proxy(url) do |callbacks|
+          callbacks.on_response do |status_code, _response|
+            ReactRenderable.log("[ReactRenderable] #{url} returned #{status_code}")
+          end
         end
+      rescue Errno::ECONNREFUSED => e
```
Comment (ismail-syed) [inline gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: Was looking into adding a test for this. How do I mock a`Errno::ECONNREFUSED` in Rails,  any examples on hand?

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb (hunk 1 above)
Comment (TheMallen) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: /shrug maybe just `[Quilt]` or something

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb (hunk 2 above)
Comment (TheMallen) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: `Errno::ECONNREFUSED.new` gives an instance of it in `irb`

Comment (TheMallen) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: so I would just mock `reverse_proxy` to throw one

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb (hunk 1 above)
Comment (ismail-syed) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: I feel like `[Quilt]` is too vague, especially to someone who might not know what quilt is who's just trying to get an app setup.

Comment (TheMallen) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: then keep it like this :p

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb (hunk 2 above)
Comment (ismail-syed) [reply gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: Opened an issue to get tests working
https://github.com/Shopify/quilt/issues/883

File: gems/quilt_rails/lib/quilt_rails/react_renderable.rb [hunk 3]
```diff
@@ -10,10 +10,14 @@ def render_react
       url = "#{Quilt.configuration.react_server_protocol}://#{Quilt.configuration.react_server_host}"
       ReactRenderable.log("[ReactRenderable] proxying to React server at #{url}")
 
-      reverse_proxy(url) do |callbacks|
-        callbacks.on_response do |status_code, _response|
-          ReactRenderable.log("[ReactRenderable] #{url} returned #{status_code}")
+      begin
```
Comment (cartogram) [inline gems/quilt_rails/lib/quilt_rails/react_renderable.rb]: TIL like try/catch for rails

**Pull request merged**